src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
 
-              deregFeatureChange = evt.feature.on('change', function(evt) {
-                var geom = evt.target.getGeometry();
-                if (geom instanceof ol.geom.Polygon) {
-                  var lineCoords = geom.getCoordinates()[0];
-
-                  if (nbPoint != lineCoords.length) {
-                    // A point is added
-                    nbPoint++;
-                  } else {
-                    var firstPoint = lineCoords[0];
-
-                    // We update features and measures
-                    var lastPoint = lineCoords[lineCoords.length - 1];
-                    var lastPoint2 = lineCoords[lineCoords.length - 2];
-
-                    var isSnapOnFirstPoint = (lastPoint[0] == firstPoint[0] &&
-                        lastPoint[1] == firstPoint[1]);
-
-                    // When the last change event is triggered the polygon is
-                    // closed so isSnapOnFirstPoint is true. We need to know
-                    // if on the change event just before, the snap on last
-                    // point was active.
-                    isFinishOnFirstPoint = (!isSnapOnLastPoint &&
-                        isSnapOnFirstPoint);
-
-                    isSnapOnLastPoint = (lastPoint[0] == lastPoint2[0] &&
-                        lastPoint[1] == lastPoint2[1]);
-
-                    if (isSnapOnLastPoint) {
-                      // In that case the 2 last points of the coordinates
-                      // array are identical, so we remove the useless one.
-                      lineCoords.pop();
-                    }
-                  }
-                  updateHelpTooltip(helpTooltip, tool.id, true,
-                      isFinishOnFirstPoint, isSnapOnLastPoint);
-                  if (tool.showMeasure) {
-                    if (!isFinishOnFirstPoint) {
-                      geom = new ol.geom.LineString(lineCoords);
-                    }
-                    updateMeasureTooltips(distTooltip, areaTooltip, geom);
-                  }
-                }
-              });
+          updateHelpTooltip(helpTooltip, tool.id, false);
+
+          if (!gaBrowserSniffer.mobile) {
+            unDrawEvts.push(map.on('pointermove', function(evt) {
+              helpTooltip.setPosition(evt.coordinate);
             }));
+          }
 
-            unDrawEvts.push(draw.on('drawend', function(evt) {
-              // Unregister the change event
-              ol.Observable.unByKey(deregFeatureChange);
-              deactivateTool(lastActiveTool);
-              scope.$applyAsync();
+          draw = new ol.interaction.Draw(tool.drawOptions);
+          var isFinishOnFirstPoint;
+          unDrawEvts.push(draw.on('drawstart', function(evt) {
+            // Remove the first features created if wanted
+            var src = layer.getSource();
+            if (tool.maxFeatures &&
+                src.getFeatures().length >= tool.maxFeatures) {
+              src.removeFeature(src.getFeatures()[0]);
```

Comment:
It was like this before but I'm not sure why this option `maxFeatures` has been added. I don't find a tool which uses that.